devlib/instrument/monsoon.py

```diff
@@ -0,0 +1,170 @@
+import devlib
+import json
+import os
+import os.path
+import psutil
+import time
+import logging
+from subprocess import Popen, STDOUT
+from threading import Timer
+import pandas as pd
+import StringIO
+from collections import namedtuple
+from subprocess import Popen, PIPE, STDOUT
+from devlib.instrument import Instrument, CONTINUOUS, MeasurementsCsv
+from devlib.exception import HostError
+from devlib.utils.misc import which
+
+class MonsoonInstrument(Instrument):
+    MONSOON_BIN = "./tools/scripts/monsoon.par"
+
+    def log_error(self, errmsg, exception=0):
+	print "ERROR: " + errmsg
+	self.logger.error(errmsg)
+	if exception == 1:
+		raise(Exception(errmsg))
+
+    def log_info(self, msg):
+	print "INFO: " + msg
+	self.logger.info(msg)
+
+    def __init__(self, target, device_entry=None):
+	super(MonsoonInstrument, self).__init__(target)
+
+	if not os.path.exists(MonsoonInstrument.MONSOON_BIN):
+		self.log_error("monsoon.par not found", exception=1)
```

Comment:
Here, if possible, we should report some tip on how to get this binary.
This is what we did in LISA for the ACME cape:
   https://github.com/ARM-software/lisa/blob/master/libs/utils/energy.py#L344
@setrofim: can this work for devlib as well?